### ISSUE TITLE
Enable SSL Passthrough

### ISSUE THREAD

**Issue by EdSalzedo** [Azure/aks-app-routing-operator#165]
Hi,

I'd like to be able to add --enable-ssl-passthrough as an argument for starting up NGINX. I currently can't find a way of doing it while using the app-routing plugin, would this be possible?

Thanks,

Ed

**Comment by OliverMKing** [comment]
Hi @EdSalzedo! Thanks for reaching out.

It's not possible to enable `ssl-passthrough` globally today but you can enable it through [Ingress annotations](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/annotations/#ssl-passthrough) on individual Ingresses.

We can look into adding a way of configuring this globally to App Routing. Tagging @sabbour so we can add it to the backlog.

Thanks!

**Comment by EdSalzedo** [comment]
Hi, thanks for responding @OliverMKing! 

I think without -enable-ssl-passthrough flag set on startup, the nginx.ingress.kubernetes.io/ssl-passthrough annotation has no effect based on the testing I've done, although I could have got that wrong.

It'd be useful to be able to configure it at some point so thank you!

**Comment by OliverMKing** [comment]
> Hi, thanks for responding @OliverMKing!
> 
> I think without -enable-ssl-passthrough flag set on startup, the nginx.ingress.kubernetes.io/ssl-passthrough annotation has no effect based on the testing I've done, although I could have got that wrong.
> 
> It'd be useful to be able to configure it at some point so thank you!

You are correct! The documentation even says it, which I missed. We will look into this.

**Comment by MXClyde** [comment]
I also have a use-case that needs this feature so forgive me a +1

**Comment by zjylo** [comment]
Another similar use case is enabling [ `-use-forwarded-headers` ](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-forwarded-headers), we're using Azure Front Door which is setting these values and then getting overwritten by Nginx.

Have to deploy manually for now till this is enabled.

**Comment by dinchand** [comment]
> Another similar use case is enabling [ `-use-forwarded-headers` ](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-forwarded-headers), we're using Azure Front Door which is setting these values and then getting overwritten by Nginx.
> 
> Have to deploy manually for now till this is enabled.

similar issue for us!

**Comment by Soren-Knudsen** [comment]
@zjylo: Can you describe your workaround? We are currently in a similar situation.

`Internet -> Azure Frontdoor -> AKS Nginx -> Container`

**Comment by sabbour** [comment]
Merged https://github.com/Azure/aks-app-routing-operator/pull/440. It will go out with the next release.

**Comment by shatteredeagle** [comment]
Hello! 
Sorry if this is the wrong place to ask but:
I see this went out in  0.2.6, but how would we update out existing installation to set the flag? Tried using az aks approuting update --enablesslpassthrough and it didn't like it, not sure if thats on the azcli or the approuting plugin? Already did the az aks approuting update to upgrade the plugin. 

Thanks!
